udf/udf_json_extract_histogram_js.sql

```diff
@@ -9,7 +9,7 @@ be more usable in practice.
 */
```

Comment:
requesting the js be used as an extension: `udf/udf_json_extract_histogram.js.sql`

Reply:
I should actually read the README carefully before making fixes :)

Reply:
To support this naming (file is `udf/udf_json_extract_histogram.js.sql` and the name of the udf itself if `udf_json_extract_histogram_js`, we need to alter the test code to replace `.` with `_` when parsing filenames, and we'll need to update the docs to specify that JS udfs should be in files ending in `.js.sql` but with names ending in `_js`.

The above seems more complicated compared to having the filename match the function name `udf_json_extract_histogram_js.sql`, so my preference is to use that convention and update the docs accordingly as part of this PR.

Alternately, we could split JS UDFs to their own `jsudf/` directory and expect that we have functions named like `jsudf_foo` defined in files named `jsudf_foo.sql`. I think this provides a nice bridge for when persistent UDFs go beta; we'd then create `udf` and `jsudf` datasets and deploy functions that are accessed like `udf.json_extract_histogram(foo)` and `jsudf.json_extract_histogram(foo)`.

What do you think?

Reply:
> The above seems more complicated compared to having the filename match the function name `udf_json_extract_histogram_js.sql`, so my preference is to use that convention and update the docs accordingly as part of this PR

this seems like a good immediate approach

> Alternately, we could split JS UDFs to their own `jsudf/` directory and expect that we have functions named like `jsudf_foo` defined in files named `jsudf_foo.sql`. I think this provides a nice bridge for when persistent UDFs go beta; we'd then create `udf` and `jsudf` datasets and deploy functions that are accessed like `udf.json_extract_histogram(foo)` and `jsudf.json_extract_histogram(foo)`.

this seems like an excellent long-term solution